lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -353,19 +348,15 @@ def sign_pki_certificate_signing_request(
             logger.warning("Error while signing PKI certificate: %s", e)
             return None
 
-    def is_pki_ca_certificate_set(self, mount: str, certificate: str) -> bool:
-        """Check if the CA certificate is set for the PKI backend."""
-        existing_certificate = self._client.secrets.pki.read_ca_certificate(mount_point=mount)
-        return existing_certificate == certificate
-
-    def create_pki_charm_role(self, role: str, allowed_domains: str, mount: str) -> None:
+    def create_or_update_pki_charm_role(self, role: str, allowed_domains: str, mount: str) -> None:
         """Create a role for the PKI backend."""
         self._client.secrets.pki.create_or_update_role(
             name=role,
             mount_point=mount,
             extra_params={
                 "allowed_domains": allowed_domains,
                 "allow_subdomains": True,
+                "allow_bare_domains": True,
```

Comment:
I don't know about adding this. This is unrelated to the bug plus I haven't seen users requesting this. I think we may only want to improve the config description to make it more obvious that only subdomains should be used.

Reply:
Alright, I removed. We can come back to it if it gets requested.
Improved the description of the config too.

Reply:
We'd still need to remove this content from the PR description

Reply:
Sure :)